base/strings/types.jl

```diff
@@ -46,6 +46,8 @@ function SubString(s::SubString, i::Int, j::Int)
     j <= endof(s) || throw(BoundsError(s, j))
     SubString(s.string, s.offset + i, s.offset + j)
 end
+SubString{T}(s::T, i::Int, j::Int) where {T<:SubString} = SubString(s, i, j)
```

Comment:
Remove this method if it's not needed (since it didn't fix the problem).

Reply:
OK - I will move it to a separate PR